pydatalab/src/pydatalab/apps/uvvis/__init__.py

```diff
@@ -0,0 +1,170 @@
+            plot_data_input = {
+                str(name): df for name, df in zip(names, absorbance_data_list)
+            }  # Ensure names are strings
+        else:
+            # Generate default names if none provided, ensuring keys are strings
+            plot_data_input = {f"Spectrum {i+1}": df for i, df in enumerate(absorbance_data_list)}
+
+        # Define the specific HoverTool configuration needed for UV-Vis
+        uv_hover_tool = HoverTool(
+            # Ensure these column names (@Wavelength, @Absorbance) exist in the DataFrames
+            tooltips=[
+                ("Wavelength / nm", "@Wavelength{0.00}"),
+                ("Absorbance", "@Absorbance{0.0000}"),
+                # Note: selectable_axes_plot doesn't automatically add the dict key/name
+                # as a column to the CDS, so we can't easily use ('Name', '@name')
+                # unless we modify selectable_axes_plot or preprocess the data.
+                # The legend will be the primary identifier.
+            ],
+            mode="vline",  # Tooltip follows the x-coordinate across all lines
+        )
+        # Call the existing selectable_axes_plot function
+        # We configure it specifically for a non-selectable UV-Vis plot use case
+        layout = selectable_axes_plot(
+            df=plot_data_input,  # Pass the list or dict of DataFrames
+            x_options=["Wavelength"],  # Fix X axis option
+            y_options=["Absorbance"],  # Fix Y axis option
+            x_default="Wavelength",  # Set default X
+            y_default="Absorbance",  # Set default Y
+            plot_points=False,  # We only want lines for spectra
+            plot_line=True,  # Ensure lines are plotted
+            tools=[uv_hover_tool],  # Add our specific hover tool
+            # color_options / color_mapper: Let selectable_axes_plot handle default color cycling
+            #                                based on index unless specific coloring logic is needed.
+            # plot_title="UV-Vis Spectra", # Optional: Add a title if desired
+            show_table=False,  # Usually don't need the table for this plot type
+        )
+        # Adding cm^-1 to the x-axis label using unicode characters - might be a more logical way
+        layout.children[0].xaxis.axis_label = "Wavelength / nm"
+        return layout
+
+    def generate_absorbance_plot(self):
+        absorbance_data = None
+        if "selected_file_order" not in self.data:
+            LOGGER.warning("No file set in the DataBlock - selected_file_order")
+            return
+
+        else:
+            file_info = []
+            for file_id in self.data["selected_file_order"]:
+                file_info.append(get_file_info_by_id((file_id), update_if_live=True))
+            # Check if the file is in the accepted file extensions
+            for file in file_info:
+                ext = os.path.splitext(file["location"].split("/")[-1])[-1].lower()
```

Comment:
This is maybe the bit that doesn't compare work with ext "raw8.txt" -- again in future we should abstract this away, but perhaps for this particular case you can also check the "-2"th extension